src/main/python/deep_qa/layers/backend/batch_dot.py

```diff
@@ -24,6 +26,69 @@ class BatchDot(Layer):
     Output:
         - a_dot_b
 
+    Examples
+    --------
+    The following examples will try to give some insight on how this layer works in relation
+    to ``K.batch_dot()``
+
+    As a first example, let's suppose that ``tensor_a`` and ``tensor_b`` have the same number
+    of dimensions. Let the shape of ``tensor_a`` be ``(2, 3, 2)``, and let the shape of ``tensor_b`` be ``(2, 4, 2)``. The mask accompanying these inputs always has one less dimension, so the
+    ``tensor_a_mask`` has shape ``(2, 3)`` and ``tensor_b_mask`` has shape ``(2, 4)``. The shape
+    of the ``batch_dot`` output would thus be ``(2, 3, 4)``. This is because we are taking the
+    batch dot of the last dimension, so the output shape is ``(2, 3)`` (from tensor_a) with ``(4)``
+    (from tensor_b) appended on (to get ``(2, 3, 4)`` in total). The output mask has the same
+    shape as the output, and is thus ``(2, 3, 4)`` as well.
+
+    >>> import keras.backend as K
+    >>> tensor_a = K.ones(shape=(2, 3, 2))
+    >>> tensor_b = K.ones(shape=(2, 4, 2))
+    >>> K.eval(K.batch_dot(tensor_a, tensor_b, axes=(2,2))).shape
+    (2, 3, 4)
+
+    Next, let's look at an example where ``tensor_a`` and ``tensor_b`` are "uneven" (different number
+    of dimensions). Let the shape of ``tensor_a`` be ``(2, 4, 2)``, and let the shape of ``tensor_b``
+    be ``(2, 4, 3, 2)``. The mask accompanying these inputs always has one less dimension, so the
+    ``tensor_a_mask`` has shape ``(2, 4)`` and ``tensor_b_mask`` has shape ``(2, 4, 3)``. The shape
+    of the ``batch_dot`` output would thus be ``(2, 4, 3)``. In the case of uneven tensors, we always
+    expand the last dimension of the smaller tensor to make them even. Thus in this case, we expand
+    ``tensor_a`` to get a new shape of ``(2, 4, 2, 1)``. Now we are taking the ``batch_dot`` of a
+    tensor with shape ``(2, 4, 2, 1)`` and ``(2, 4, 3, 2)``. Note that the first two dimensions of
+    this tensor are the same ``(2, 4)`` -- this is an requirement imposed by ``K.batch_dot``.
```

Comment:
Also, add a note that this is _different_ from the documented behavior of `K.batch_dot` - the documentation is wrong.